### DISCUSSION TITLE
Translate a selection

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#148] [Show and tell]
I just created a simple shell command for opening Google Translate in a web browser, passing it the text that is selected in Obsidian.

![kuva](https://user-images.githubusercontent.com/13002142/152853380-77307a40-563b-4bb3-8a8e-5e1a4df1e8a2.png)

- The shell command on Windows PowerShell: `start "https://translate.google.fi/?hl=fi&sl=auto&tl=fi&op=translate&text={{selection}}"` (note that you need to change `tl=fi` to your own language, e.g `tl=en` for English or `tl=pt` for Portuguese). Source language is `auto`, i.e. detected automatically.
- To make it available in the right click context menu in the editor, enable the _Editor menu_ event:
    ![kuva](https://user-images.githubusercontent.com/13002142/152853950-dea617b6-95c1-4fae-b8ef-b4819da1d040.png)

---

Things to develop further:
- As SC does not yet have an ability to do URL encoding (#95), it's possible that some text might break the URL. E.g. if the selected text would contain `&`, then I guess all text after `&` would be left out from the translation.
- Instead of opening a web browser, use [Google Translate API](https://cloud.google.com/translate) to fetch the translation and then display it in [an output modal](https://github.com/Taitava/obsidian-shellcommands/issues/134#issuecomment-1003641475).

---

P.S. One translator plugin was mentioned in #92.

**Comment by FelipeRearden** [comment]
Thank you very much @Taitava !!!!!

Great idea for us learning new languages .

Note to myself: search about MacOS dictionary Apple script to open dictionary app using {{selection}} as input
